src/main/java/Command.java

```diff
@@ -0,0 +1,8 @@
+/**
+ * Abstract class for handling different commands.
+ */
+public abstract class Command {
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage)
+            throws DoneException, DeleteException, DeadlineException, EventException, TodoException, FindException;
```

Comment:
I noticed that DeadlineException and EventException extends TaskException but the rest extend Exception. Would it be better if all your relevant custom exceptions extend a custom parent exception (e.g. DukeException) so that you can change this method to "throws DukeException" instead?

Reply:
That's a good idea. Thank you for your suggestion.